modules/db/src/test/java/org/mule/module/db/internal/domain/statement/QueryStatementFactoryTestCase.java

```diff
@@ -29,18 +33,43 @@
 {
 
     @Test
-    public void createsPreparedStatements() throws Exception
+    public void createsPreparedStatementsWithScrolling() throws Exception
     {
 
         String sqlText = "call test";
         CallableStatement createdStatement = mock(CallableStatement.class);
+        DatabaseMetaData metadata = mock(DatabaseMetaData.class);
         DbConnection connection = mock(DbConnection.class);
-        when(connection.prepareCall(sqlText, ResultSet.TYPE_FORWARD_ONLY, ResultSet.CONCUR_READ_ONLY)).thenReturn(createdStatement);
+
+        when(metadata.supportsResultSetType(TYPE_SCROLL_INSENSITIVE)).thenReturn(true);
+        when(connection.getMetaData()).thenReturn(metadata);
+        when(connection.prepareCall(sqlText, TYPE_SCROLL_INSENSITIVE, CONCUR_READ_ONLY)).thenReturn(createdStatement);
+
+        QueryStatementFactory factory = new QueryStatementFactory();
+        QueryTemplate queryTemplate = new QueryTemplate(sqlText, STORE_PROCEDURE_CALL, Collections.<QueryParam>emptyList());
+        CallableStatement statement = (CallableStatement) factory.create(connection, queryTemplate);
+
+        assertThat(statement, is(createdStatement));
+    }
+
+    @Test
+    public void createsPreparedStatementsWithForwardOnly() throws Exception
+    {
+
+        String sqlText = "call test";
+        CallableStatement createdStatement = mock(CallableStatement.class);
```

Comment:
Looks like the same set of mocks are used on all the tests. Can you declare them as fields to remove cod duplication?